### ISSUE TITLE
NPE on applying branding

### ISSUE THREAD

**Issue by MediaPlusPL** [stefan-niedermann/nextcloud-deck#782]
Version 1.13.5
Telephone huawei Mate 10 lite

```
Full Crash:

App Version: 1.13.5
App Version Code: 1013005
App Flavor: play

Files App Version Code: 30140190

---

OS Version: 4.4.23+(354(C432))
OS API Level: 26
Device: HWRNE
Manufacturer: HUAWEI
Model (and Product): RNE-L21 (RNE-L21)

---

java.lang.RuntimeException: Unable to start activity ComponentInfo{it.niedermann.nextcloud.deck.play/it.niedermann.nextcloud.deck.ui.card.EditActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void it.niedermann.nextcloud.deck.ui.card.attachments.CardAttachmentAdapter.applyBrand(int)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3303)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3411)
	at android.app.ActivityThread.-wrap12(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1994)
	at android.os.Handler.dispatchMessage(Handler.java:108)
	at android.os.Looper.loop(Looper.java:166)
	at android.app.ActivityThread.main(ActivityThread.java:7529)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void it.niedermann.nextcloud.deck.ui.card.attachments.CardAttachmentAdapter.applyBrand(int)' on a null object reference
	at it.niedermann.nextcloud.deck.ui.card.attachments.CardAttachmentsFragment.applyBrand(CardAttachmentsFragment.java:518)
	at it.niedermann.nextcloud.deck.ui.branding.BrandedFragment.onStart(BrandedFragment.java:19)
	at androidx.fragment.app.Fragment.performStart(Fragment.java:2731)
	at androidx.fragment.app.FragmentStateManager.start(FragmentStateManager.java:365)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1206)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1368)
	at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1446)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1509)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2637)
	at androidx.fragment.app.FragmentManager.dispatchStart(FragmentManager.java:2595)
	at androidx.fragment.app.FragmentController.dispatchStart(FragmentController.java:258)
	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:550)
	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:210)
	at it.niedermann.nextcloud.deck.ui.branding.BrandedActivity.onStart(BrandedActivity.java:22)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1339)
	at android.app.Activity.performStart(Activity.java:7403)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3266)
	... 9 more

```

**Comment by desperateCoder** [comment]
Hi @MediaPlusPL, thanks for your report!

We'll have a look at this, sorry for the inconveniences.

One thing I'd also like to mention: The term "Unstable application" in the title sounds a little bit like an offense. We'll fix this as soon as possible, but reading this - while it works for 99% of our users - just hurts. We do our best providing a stable application, but we also do this in our spare time, and we are just two random guys most of the time, so please be a little bit more polite please. It's just impossible to test our app on each and every constellation of devices and server-configurations out there.

Also we have an issue template (see [here](https://github.com/stefan-niedermann/nextcloud-deck/blob/master/FAQ.md#why-has-my-bug-report-been-closed)), please fill it out the next time you experience some difficulties. This provides some useful informations for us to be able to help you much faster.

As already said: we'll have a look at this. Stay safe!

**Comment by MediaPlusPL** [comment]
I have next problem :(

App Version: 1.13.5
App Version Code: 1013005
Server App Version: 1.2.2
App Flavor: play

Files App Version Code: 30140190

---

OS Version: 4.4.23+(354(C432))
OS API Level: 26
Device: HWRNE
Manufacturer: HUAWEI
Model (and Product): RNE-L21 (RNE-L21)

---

java.lang.ArrayIndexOutOfBoundsException: length=22; index=22
	at java.util.ArrayList.add(ArrayList.java:468)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.CardDataProvider$1.onResponse(CardDataProvider.java:53)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.CardDataProvider$1.onResponse(CardDataProvider.java:50)
	at it.niedermann.nextcloud.deck.api.RequestHelper$ResponseConsumer.accept(RequestHelper.java:62)
	at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:62)
	at io.reactivex.internal.operators.observable.ObservableFromPublisher$PublisherSubscriber.onNext(ObservableFromPublisher.java:56)
	at com.nextcloud.android.sso.api.NextcloudAPI.lambda$performRequestObservableV2$1$NextcloudAPI(NextcloudAPI.java:111)
	at com.nextcloud.android.sso.api.-$$Lambda$NextcloudAPI$8WS9mapJbOYsPgbhBi25vBNYZqI.subscribe(Unknown Source:6)
	at io.reactivex.internal.operators.observable.ObservableFromPublisher.subscribeActual(ObservableFromPublisher.java:31)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableMap.subscribeActual(ObservableMap.java:32)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at java.lang.Thread.run(Thread.java:784)

```

**Comment by desperateCoder** [comment]
@MediaPlusPL sorry for the inconveniences! We'll have a loo at it in a dedicated issue: https://github.com/stefan-niedermann/nextcloud-deck/issues/789

**Comment by stefan-niedermann** [comment]
Will be fixed in `1.14.1`.
